### PR TITLE
feat: RPG-130 added GUI config loading

### PR DESCRIPTION
**Pull request by co012**
# Description
Made it possible to choose config using GUI.

To bypass config selection add config parameter to run task.
example: `run --args="--config="C:\Dev\rpg-io\configurations\demo-config-1""`

# Checklist

* [ ] Included code to test these changes
* [ ] Updated Jira

### CODE REVIEW

File: src/main/java/io/rpg/wrapper/ConfigChooser.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  private final FileChooser chooser;
+
+  public ConfigChooser() {
+    chooser = new FileChooser();
+    chooser.setTitle("Select root.json");
+    chooser.getExtensionFilters().addAll(
+        new FileChooser.ExtensionFilter("config", "root.json"),
+        new FileChooser.ExtensionFilter("json", "*.json")
+    );
+  }
```
Comment (kkafar) [inline src/main/java/io/rpg/wrapper/ConfigChooser.java:18]: I believe `ConfigLoader` actually expects the path to the directory that contains configuration (including `root.json`), not to `root.json` file.

File: src/main/java/io/rpg/wrapper/ConfigChooser.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   * @param stage - main stage.
+   * @return Optional of selected file folder path.
+   */
+  public Optional<String> open(Stage stage) {
+    File file = chooser.showOpenDialog(stage);
+    if (file == null) {
+      return Optional.empty();
+    }
+
+    return Optional.of(file.getParentFile().getAbsolutePath());
```
Comment (kkafar) [inline src/main/java/io/rpg/wrapper/ConfigChooser.java:32]: Oh, you're resolving it here? Why just not let the user pick the directory?

File: src/main/java/io/rpg/wrapper/WrapperController.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  private void onStartClick() {
+    startButton.setDisable(true);
+    game.start(stage);
+  }
+
+  private void printLine(String line) {
+    outputArea.setText(outputArea.getText() + "\n" + line);
+  }
+
+
```
Comment (kkafar) [inline src/main/java/io/rpg/wrapper/WrapperController.java:149]:
```suggestion
```

File: src/main/java/io/rpg/wrapper/WrapperController.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      startButton.setDisable(true);
+      return;
+    }
+
+    String path = pathOptional.get();
+    printLine("Selected: " + path);
+    startButton.setDisable(false);
+
+    loadGame(path);
+    game.setOnEnd(() -> show(stage));
```
Comment (kkafar) [inline src/main/java/io/rpg/wrapper/WrapperController.java:108]: It look like when `loadGame` fails (`game` field is not initialized) then we crash by `NullPointerException` in line 108. Maybe make `loadGame` return the `Result` and react to it here?
```suggestion
    loadGame(path).ifOkOrElse(ok -> { game = ok; game.setOnEnd... }, err -> { /* noop  or handle the error ? */}
```

**Review by kkafar — CHANGES_REQUESTED** [review]

File: src/main/java/io/rpg/wrapper/WrapperController.java (hunk 4 above)
Comment (co012) [reply src/main/java/io/rpg/wrapper/WrapperController.java]: :thumbsup:

**Review by kkafar — APPROVED** [review]
LGTM

**Pull request merged**